data_processing/common/GraphEnum.py

```diff
@@ -1,38 +1,256 @@
+		> Connecting to: neo4j://localhost:7687
+		> Connection successfull: True
+		> Running on: localhost
+		> Lookup ID: 1.2.840...
+		> Found: [<Record id(container)=7091 labels(container)=['scan'] container.type='scan' container.name='1.2.840...>]
+		> Match on: WHERE id(container) = 7091 
+		> Successfully attached to: scan 1.2.840...
+
+	$ node = Node("dicom", "DCM-0123", {"Modality":"CT", "path":"file:/some/path/1.dcm"})
+
+	$ container.add(node)
+		> Adding: test-0000
+		  Container has 1 pending commits
+
+	$ container.saveAll()
+		> Committing dicom:globals{  name: 'DCM-0123', QualifiedPath: 'test::DCM-0123', Namespace: 'test', type: 'dicom' , path: 'file:/some/path/1.dcm'}
+
+	$ node = container.ls("dicom")
+		> ----------------------------------------------------------------------------------------------------
+		  name: DCM-0123
+		  type: dicom
+		  properties: 
+		  - type: 'dicom'
+		  - QualifiedPath: 'test::DCM-0123'
+		  - path: 'file:/some/path/1.dcm'
+		  - Namespace: '3'
+		  - Modality: 'CT'
+		  - name: 'DCM-0123'
+		  ----------------------------------------------------------------------------------------------------
+	$ container.get("dicom", "data.Namespace='test'").path
+		> /some/path/1.dcm
+
+	$ container.get("dicom", "data.Namespace='test'").properties['Modality']
+		> 'CT'
+
+	
 	"""
-	def __init__(self, conn, container_id):
+	# TODO: worried about schema issues? like making sure name, namespace, type and qualified path are present, neo4j offers schema enforcment. 
+	# TODO: testing
+	# TODO: error checking
+
+	def __init__(self, params):
+		"""
+		Initialize the container object.
+			Connects to the graph DB
+			Figure out what host this code is running on
+
+		:params: params - dictonary of important configuration, right now, only the graph URI connection parameters are needed.
+		"""
+
+		self.params=params
+
+		# Connect to graph DB
+		print ("Connecting to:", params['GRAPH_URI'])
+		self._conn = Neo4jConnection(uri=params['GRAPH_URI'], user=params['GRAPH_USER'], pwd=params['GRAPH_PASSWORD'])
+		print ("Connection successfull:", self._conn.test_connection())
+		self._host = socket.gethostname() # portable to *docker* containers
+		print ("Running on:", self._host)
+	
+	def setNamespace(self, namespace_id: str):
+		"""
+		Sets the namespace for this container's commits
+
+		:params: namespace_id - namespace value 
+		"""
+		self._namespace_id = namespace_id
+		return self
+	
+	def lookupAndAttach(self, container_id):
+		"""
+		Checks if the node referenced by container_id is a valid container, queries the metastore for relevant metadata
+
+		:params: container_id - the unique container ID, either as an integer (neo4j autopopulated ID) or as a string (the Qualified Path)
+		"""
+		self._attached = False
+		print ("Lookup ID:", container_id)
+
+		# Figure out how to match the node
+		if type(container_id) is str: 
```

Comment:
It'd be good to add a check for the string containing "::", as additional validation that the string is a QualifiedPath.